src/java/main/org/apache/zookeeper/server/quorum/QuorumPeerConfig.java

```diff
@@ -292,6 +293,8 @@ public void parseProperties(Properties zkProp)
                 }
             } else if ((key.startsWith("server.") || key.startsWith("group") || key.startsWith("weight")) && zkProp.containsKey("dynamicConfigFile")) {
                 throw new ConfigException("parameter: " + key + " must be in a separate dynamic config file");
+            } else if(key.equals("warn.responseTime")) {
```

Comment:
This is a new property, so it should be:

* Documented - in particular mentions that this log can be disabled if set to -1. We don't want user to check code to figure out how to use this.
* Add a test case to cover various code paths.

Also, the name used here could be improved. I think something like request.warningthresholdms is more accurate. We already have fsync.warningthresholdms which in many ways has similar functionality of what's being added, so use that as a reference.